wrangler-core/src/main/java/co/cask/wrangler/utils/Json2Schema.java

```diff
@@ -108,12 +108,17 @@ public Schema toSchema(String id, Row row) throws RecordConvertorException {
       }
 
       // TODO - remove all the instaces of java.util.Date once all the directives support LogicalType.
-      if (value instanceof Date || value instanceof java.sql.Date || value instanceof Time
+      if (value instanceof java.sql.Date || value instanceof Time
```

Comment:
why we setting it to Long in these cases if not doing for `Date`?

Reply:
this also seems to be impacting other directives and not just parse-as-excel. Have we considered what all directives this change will impact?

Reply:
I returned Date only for Date use case else cases not touched . And in latest build of guavus cdap we have supported Date data type . In all directive it will be returned as date . if any sink supports date type then it works .